optuna/samplers/_tpe/sampler.py

```diff
@@ -742,52 +750,30 @@ def _calculate_weights_below_for_multi_objective(
     below_trials: list[FrozenTrial],
     constraints_func: Callable[[FrozenTrial], Sequence[float]] | None,
 ) -> np.ndarray:
-    loss_vals = []
-    feasible_mask = np.ones(len(below_trials), dtype=bool)
-    for i, trial in enumerate(below_trials):
-        # Hypervolume contributions are calculated only using feasible trials.
-        if constraints_func is not None:
-            if any(constraint > 0 for constraint in constraints_func(trial)):
-                feasible_mask[i] = False
-                continue
-        values = []
-        for value, direction in zip(trial.values, study.directions):
-            if direction == StudyDirection.MINIMIZE:
-                values.append(value)
-            else:
-                values.append(-value)
-        loss_vals.append(values)
-    lvals = np.asarray(loss_vals, dtype=float)
-
-    # Calculate weights based on hypervolume contributions.
-    n_below = len(lvals)
-    weights_below: np.ndarray
-    if n_below == 0:
-        weights_below = np.asarray([])
-    elif n_below == 1:
-        weights_below = np.asarray([1.0])
-    else:
-        worst_point = np.max(lvals, axis=0)
-        reference_point = np.maximum(1.1 * worst_point, 0.9 * worst_point)
-        reference_point[reference_point == 0] = EPS
-        hv = compute_hypervolume(lvals, reference_point)
-        indices_mat = ~np.eye(n_below).astype(bool)
-        contributions = np.asarray(
-            [
-                hv - compute_hypervolume(lvals[indices_mat[i]], reference_point)
-                for i in range(n_below)
-            ]
-        )
-        contributions[np.isnan(contributions)] = np.inf
-        max_contribution = np.maximum(np.max(contributions), EPS)
-        if not np.isfinite(max_contribution):
-            weights_below = np.ones_like(contributions, dtype=float)
-            # TODO(nabenabe0928): Make the weights for non Pareto solutions to zero.
-            weights_below[np.isfinite(contributions)] = EPS
```

Comment:
As mentioned here,
- https://github.com/optuna/optuna/pull/5871#issue-2749413723

if `hv` is `inf`, all the elements in `contributions` also become `inf`.

Let's define this array as `hvs`:

```python
hvs = np.array([compute_hypervolume(lvals[indices_mat[i]], reference_point) for i in range(n_below)])
```

If `hvs[i]` is `inf`, `hv - hvs[i]` becomes `nan`, leading to `contributions[i] = np.inf` due to `contributions[np.isnan(contributions)] = np.inf`.

If `hvs[i]` is finite, `hv - hvs[i]` anyways becomes `inf`, so `contributions[i] = np.inf`.

It simply means that if `hv` is `inf`, all the elements in `contributions` are `inf`.

That is why `np.isfinite(contributions)` is all False, so all the elements in `weights_below` are 1.